src/scripts/create_hunter_csv.py

```diff
@@ -0,0 +1,477 @@
+                    Whether the analysis is prospective (True by default).
+
+    Returns:
+            A tuple with the dataframe with performance results and the corresponding test type.
+    """
+    hunter_df_out = generate_hunter_df(json_paths, is_prospective)
+
+    if len(json_paths) == 0 or hunter_df_out.empty:
+        logging.error(
+            "Either the 'json_paths' or the hunter dataframe is empty; "
+            "thus, empty dataframe and string are being returned."
+        )
+        return pd.DataFrame(), ''
+
+    test_type = json_paths[0].split(os.sep)[5]
+
+    if test_type != '' and not hunter_df_out.empty:
+
+        return hunter_df_out, test_type
+
+
+if __name__ == '__main__':
+
+    # cd into nightly_results
+    nightly_result_path = NIGHTLY_RESULTS_DIR
+    cd_into_proj_dir(nightly_result_path)
+
+    # Get sorted list of dates from output folders with performance tests
+    nightly_result_dates = os.listdir(nightly_result_path)
+    nightly_result_dates.sort()
+
+    # Set to False if running this for the first time, then re-run and set to True.
+    is_case_prospective = PROSPECTIVE_MODE
+    if is_case_prospective:
+        # cd into hunter_csv
+        cd_into_proj_dir()
+
+        hunter_df_fixed_100 = pd.read_csv(
+            f'{FIXED_100_CSV_NAME}{HUNTER_FILE_FMT}')
+        hunter_df_rated_100 = pd.read_csv(
+            f'{RATED_100_CSV_NAME}{HUNTER_FILE_FMT}')
+        hunter_df_fixed_1000 = pd.read_csv(
+            f'{FIXED_1000_CSV_NAME}{HUNTER_FILE_FMT}')
+        hunter_df_rated_1000 = pd.read_csv(
+            f'{RATED_1000_CSV_NAME}{HUNTER_FILE_FMT}')
+        hunter_df_fixed_10000 = pd.read_csv(
+            f'{FIXED_10000_CSV_NAME}{HUNTER_FILE_FMT}')
+        hunter_df_rated_10000 = pd.read_csv(
+            f'{RATED_10000_CSV_NAME}{HUNTER_FILE_FMT}')
+
+        # Get date from the latest test run
+        test_input_date = nightly_result_dates[-1]
+
+        # Separated by _ to be compared wrt 'test_input_date'
+        last_date_from_csv = hunter_df_fixed_100['time'].iloc[-1].split(' ')[0].replace('-', '_')
+
+        if test_input_date == last_date_from_csv:
+            raise ValueError(f"The test results for the date '{test_input_date}' were already run and summarised "
+                             f"in the csv file. Please ensure the date considered is not in the csv file and it is "
+                             f"past the latest date in the csv file.")
+        type_of_test = type(test_input_date)
+
+        yesterday_s_date = get_yesterday_date(FMT_Y_M_D)
+
+        # Get path to the latest test run
+        path_w_date = f'{nightly_result_path}{os.sep}{test_input_date}'
+
+        # cd into filename with a date-specific test run (path_w_date)
+        cd_into_proj_dir(path_w_date)
+
+        path_to_each_test_json = get_paths_to_six_json(path_w_date)
+
+        list_of_hunter_df = []
+        list_of_type_of_tests = []
+        for test_json_path in path_to_each_test_json:
+
```

Comment:
Can you delete the empty line after `if` and `for`, just for consistency across the file.